packages/grafana-ui/src/components/Forms/Legacy/Select/ButtonSelect.tsx

```diff
@@ -1,101 +1,81 @@
-  isMenuOpen?: boolean;
-  onOpenMenu?: () => void;
-  onCloseMenu?: () => void;
-  tabSelectsValue?: boolean;
-  autoFocus?: boolean;
+  narrow?: boolean;
+  variant?: ButtonVariant;
 }
 
-export class ButtonSelect<T> extends PureComponent<Props<T>> {
-  onChange = (item: SelectableValue<T>) => {
-    const { onChange } = this.props;
+/** @internal */
+export const ButtonSelect = React.memo(<T,>(props: Props<T>) => {
+  const { className, options, value, onChange, narrow, variant } = props;
+  const [isOpen, setIsOpen] = useState<boolean>(false);
+  const styles = useStyles(getStyles);
+
+  const onCloseMenu = () => {
+    setIsOpen(false);
+  };
+
+  const onToggle = (event: React.MouseEvent) => {
+    event.stopPropagation();
+    event.preventDefault();
+    setIsOpen(!isOpen);
+  };
+
+  const onChangeInternal = (item: SelectableValue<T>) => {
     onChange(item);
+    setIsOpen(false);
   };
 
-  render() {
-    const {
-      className,
-      options,
-      value,
-      label,
-      iconClass,
-      components,
-      maxMenuHeight,
-      tooltipContent,
-      isMenuOpen,
-      onOpenMenu,
-      onCloseMenu,
-      tabSelectsValue,
-      autoFocus = true,
-    } = this.props;
+  const menuGroup: MenuItemsGroup = {
+    items: options.map((item) => ({
+      label: (item.label || item.value) as string,
+      onClick: () => onChangeInternal(item),
+      active: item.value === value?.value,
+    })),
+  };
 
-    const combinedComponents = {
-      ...components,
-      Control: ButtonComponent({ label, className, iconClass }),
-    };
+  return (
+    <>
+      <ToolbarButton className={className} isOpen={isOpen} onClick={onToggle} narrow={narrow} variant={variant}>
+        {value?.label || value?.value}
+      </ToolbarButton>
+      {isOpen && (
+        <div className={styles.menuWrapper}>
+          <ClickOutsideWrapper onClick={onCloseMenu} parent={document}>
+            <Menu items={[menuGroup]} />
+          </ClickOutsideWrapper>
+        </div>
+      )}
+    </>
+  );
+});
 
-    return (
-      <Select
-        autoFocus={autoFocus}
-        backspaceRemovesValue={false}
-        isClearable={false}
-        isSearchable={false}
-        options={options}
-        onChange={this.onChange}
-        value={value}
-        isOpen={isMenuOpen}
-        onOpenMenu={onOpenMenu}
-        onCloseMenu={onCloseMenu}
-        maxMenuHeight={maxMenuHeight}
-        components={combinedComponents}
-        className="gf-form-select-box-button-select"
-        tooltipContent={tooltipContent}
-        tabSelectsValue={tabSelectsValue}
-      />
-    );
-  }
-}
+ButtonSelect.displayName = 'ButtonSelect';
+
+const getStyles = (theme: GrafanaTheme) => {
+  return {
+    wrapper: css`
+      position: relative;
+      display: inline-flex;
+    `,
+    menuWrapper: css`
+      position: absolute;
+      z-index: ${theme.zIndex.dropdown};
+      top: ${theme.spacing.formButtonHeight + 1}px;
+      right: 0;
```

Comment:
Should this be `left` instead? Otherwise the dropdown is not aligned (at leat in storybook).

Reply:
no, the only uses we have right now should be right aligned, have updated the story